plugins/dev9ghzdrk/Linux/Linux.cpp

```diff
@@ -0,0 +1,142 @@
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+
+#include <gtk/gtk.h>
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <string.h>
+
+#include "Config.h"
+#include "../DEV9.h"
+#include "pcap.h"
+#include "pcap_io.h"
+#include "net.h"
+
+static GtkBuilder * builder;
+
+void SysMessage(char *fmt, ...) {
+    va_list list;
+    char tmp[512];
+
+    va_start(list,fmt);
+    vsprintf(tmp,fmt,list);
+    va_end(list);
+
+    GtkWidget *dialog = gtk_message_dialog_new (NULL,
+                        GTK_DIALOG_MODAL,
+                        GTK_MESSAGE_ERROR,
+                        GTK_BUTTONS_CLOSE,
+                        tmp);
+    gtk_dialog_run (GTK_DIALOG (dialog));
+    gtk_widget_hide(dialog);
+}
+
+void OnInitDialog() {
+    char *dev;
+    gint idx = 0;
+
+    LoadConf();
+
+    gtk_combo_box_text_append_text((GtkComboBoxText *)gtk_builder_get_object(builder,"IDC_BAYTYPE"),"Expansion");
+    gtk_combo_box_text_append_text((GtkComboBoxText *)gtk_builder_get_object(builder,"IDC_BAYTYPE"),"PC Card");
+    for (int j=0;j<2;j++)
+    {
+    for (int i=0; i<pcap_io_get_dev_num(); i++) {
+        dev = pcap_io_get_dev_name(i,j);
+        if( 'b' != dev[5] )
+            continue;
+        gtk_combo_box_text_append_text((GtkComboBoxText *)gtk_builder_get_object(builder,"IDC_ETHDEV"), dev);
+        if (strcmp(dev, config.Eth) == 0) {
+        gtk_combo_box_set_active((GtkComboBox *)gtk_builder_get_object(builder,"IDC_ETHDEV"),idx);
+        }
+        idx++;
+    }
+    }
+    gtk_entry_set_text ((GtkEntry *)gtk_builder_get_object(builder,"IDC_HDDFILE"), config.Hdd);
+    gtk_toggle_button_set_active ((GtkToggleButton *)gtk_builder_get_object(builder,"IDC_ETHENABLED"),
+                  config.ethEnable);
+    gtk_toggle_button_set_active ((GtkToggleButton *)gtk_builder_get_object(builder,"IDC_HDDENABLED"),
+                  config.hddEnable);
+
+}
+
+void OnOk() {
+
+    char* ptr = gtk_combo_box_text_get_active_text((GtkComboBoxText *)gtk_builder_get_object(builder,"IDC_ETHDEV"));
+    strcpy(config.Eth, ptr);
+
+    strcpy(config.Hdd, gtk_entry_get_text ((GtkEntry *)gtk_builder_get_object(builder,"IDC_HDDFILE")));
+
+    config.ethEnable = gtk_toggle_button_get_active ((GtkToggleButton *)gtk_builder_get_object(builder,"IDC_ETHENABLED"));
+    config.hddEnable = gtk_toggle_button_get_active ((GtkToggleButton *)gtk_builder_get_object(builder,"IDC_HDDENABLED"));
+
+    SaveConf();
+
+}
+
+EXPORT void DEV9configure() {
+    GtkDialog *dlg = GTK_DIALOG (gtk_builder_get_object(builder, "IDD_CONFDLG"));
+    OnInitDialog();
+    gint result = gtk_dialog_run (dlg);
+    switch(result) {
+    case -5: //IDOK
+    OnOk();
+    break;
+    case -6: //IDCANCEL
+    break;
+    }
+    gtk_widget_hide (GTK_WIDGET(dlg));
+
+}
+
+void __attribute__((constructor)) DllMain() {
+
+    gtk_init (NULL, NULL);
+
+    builder = gtk_builder_new();
+    gtk_builder_add_from_file(builder, "dev9ghzdrk.ui", NULL);
```

Comment:
And, if necessary, perform a Copy from whatever that reasonable DESTINATION directory was to the directory stored in s_strIniPath

Reply:
still needs attention

Reply:
Would it be possible to embed the ui file into the plugin? Either using objcopy or GResource

I found an article discussing using GResouce with gtk2 here https://metzlog.srcbox.net/2014/11/embedding-gtkbuilder-files-in-gtk2/